src/auto-evo/selection_pressure/SelectionPressure.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace AutoEvo;
+
+using System.Collections.Generic;
+
+public abstract class SelectionPressure
+{
+    public readonly float Strength;
+    public readonly List<IMutationStrategy<MicrobeSpecies>> Mutations;
+
+    public SelectionPressure(float strength, List<IMutationStrategy<MicrobeSpecies>> mutations)
+    {
+        Strength = strength;
+        Mutations = mutations;
+    }
+
+    public abstract LocalizedString Name
+    {
+        get;
+    }
+
+    public abstract float Score(Species species, Patch patch, SimulationCache cache);
+    public abstract float GetEnergy(Patch patch);
+
+    /// <summary>
+    ///   Calculates the relative difference between the old and new scores
+    /// </summary>
+    public float WeightedComparedScores(float newScore, float oldScore)
+    {
+        if (newScore <= 0)
+        {
+            return -1;
+        }
+
+        if (oldScore == 0)
+        {
+            return newScore > 0 ? 1 : 0;
+        }
+
+        if (newScore > oldScore)
+        {
+            return newScore / oldScore * Strength;
+        }
+
+        if (oldScore > newScore)
+        {
+            return -(oldScore / newScore) * Strength;
+        }
+
+        return 0;
+    }
+
+    /// <summary>
+    ///   A description of this miche.
+    /// </summary>
+    public virtual LocalizedString GetDescription()
+    {
+        return Name;
+    }
+
+    // ToString is used to display the Selection Pressure in the Miche Tree
+    public override string ToString()
```

Comment:
Why is ToString used for displaying to the user? This seems a bit bad to me as elsewhere in Thrive I use ToString purely for easier to see values in the debugger.